rust/noosphere-core/src/api/client.rs

```diff
@@ -529,3 +567,20 @@ where
         Ok(push_response)
     }
 }
+
+#[cfg(feature = "test-gateway")]
+fn apply_test_header(headers: &mut HeaderMap, forced_host_header: &reqwest::header::HeaderValue) {
+    use reqwest::header::HOST;
+    _ = headers.remove(HOST);
+    headers.insert(HOST, forced_host_header.to_owned());
+}
+
+#[cfg(feature = "test-gateway")]
+fn create_test_header(identity: &str) -> Result<reqwest::header::HeaderValue> {
+    Ok(reqwest::header::HeaderValue::from_str(&format!(
+        "{}.gateway.test",
```

Comment:
Would it be possible scope the subdomain to the host of the configured gateway URL e.g., if user configured `http://localhost:8493` as their gateway URL, the host becomes `$SPHERE_ID.localhost`?